src/compass/s1_cslc_qa.py

```diff
@@ -466,3 +481,133 @@ def write_qa_dicts_to_json(self, file_path):
+
+            Parameters
+            ----------
+            cslc_h5py_path: h5py.File
+                Root of the CSLC-S1 HDF5 product
+            shapefile_path: str
+                Path to the coastline shapefile
+
+
+            Returns
+            -------
+            ratio_valid_pixel_land:  float
+                Ratio of valid pixels on land
+        '''
+
+        drv_shp_in = ogr.GetDriverByName('GPKG')
+        if os.path.exists(LAND_GPKG_FILE):
+            coastline_shapefile = drv_shp_in.Open(LAND_GPKG_FILE)
+        else:
+            raise RuntimeError(f'cannot find land polygon file: {LAND_GPKG_FILE}')
+
+        layer_coastline = coastline_shapefile.GetLayer()
+
+        # extract the geogrid information
+        epsg_cslc = int(cslc_h5py_root[f'{DATA_PATH}/projection'][()])
+
+        x_spacing = float(cslc_h5py_root[f'{DATA_PATH}/x_spacing'][()])
+        y_spacing = float(cslc_h5py_root[f'{DATA_PATH}/y_spacing'][()])
+
+        x0 = list(cslc_h5py_root[f'{DATA_PATH}/x_coordinates'][()])[0] - x_spacing / 2
+        y0 = list(cslc_h5py_root[f'{DATA_PATH}/y_coordinates'][()])[0] - y_spacing / 2
+
+        cslc_array = np.array(cslc_h5py_root[f'{DATA_PATH}/{pol}'])
+
+        height_cslc, width_cslc = cslc_array.shape
+
+        drv_raster_out = gdal.GetDriverByName('MEM')
+        rasterized_land = drv_raster_out.Create(str(time.time_ns),
+                                                width_cslc, height_cslc,
+                                                1, gdal.GDT_Byte)
+        srs_raster = osr.SpatialReference()
+        srs_raster.ImportFromEPSG(epsg_cslc)
+
+        rasterized_land.SetGeoTransform((x0, x_spacing, 0, y0, 0, y_spacing))
+        rasterized_land.SetProjection(srs_raster.ExportToWkt())
+
+        gdal.RasterizeLayer(rasterized_land, [1], layer_coastline)
+
+        mask_land = rasterized_land.ReadAsArray()
+
+        mask_geocoded_burst = _get_valid_pixel_mask(cslc_array)
+
+        mask_valid_inside_burst = mask_geocoded_burst & ~np.isnan(cslc_array)
+
+        mask_valid_land_pixel = mask_geocoded_burst & mask_land
+
+        percent_valid_land_px = mask_valid_land_pixel.sum() / mask_geocoded_burst.sum() * 100
+        percent_valid_px = mask_valid_inside_burst.sum() / mask_geocoded_burst.sum() * 100
+
+        return percent_valid_land_px, percent_valid_px
+
+
+
+
+    def compute_layover_shadow_pixel_percent(self, cslc_h5py_root):
+        '''
+            Compute the ratio of valid pixels on land area
+
+            Parameters
+            ----------
+            cslc_h5py_path: h5py.File
+                Root of the CSLC-S1 HDF5 product
+            shapefile_path: str
+                Path to the coastline shapefile
+
+
+            Returns
+            -------
+            ratio_valid_pixel_land:  float
+                Ratio of valid pixels on land
+        '''
+
+        layover_shadow_mask_array = np.array(cslc_h5py_root[f'{DATA_PATH}/layover_shadow_mask'])
```

Comment:
```suggestion
        layover_shadow_mask_array = cslc_h5py_root[f'{DATA_PATH}/layover_shadow_mask'][()]
```
`[()]` should return `np.ndarray`?

Reply:
Yes it does. Thanks for the suggestion!